### PR TITLE
MBS-10042: Port all Aliases tabs to React

### PR DESCRIPTION
**Pull request by reosarevok**
https://tickets.metabrainz.org/browse/MBS-10042

### CODE REVIEW

File: root/components/Aliases/index.js [hunk 1]
```diff
@@ -15,14 +15,30 @@ import EntityLink from '../../static/scripts/common/components/EntityLink';
...
+        return !$c.user.is_editing_disabled;
+    }
+  }
+  return false;
+}
+
 type Props = {
   +$c: CatalystContextT,
   +aliases: $ReadOnlyArray<AliasT>,
   +allowEditing?: boolean,
```
Comment (mwiencek) [inline root/components/Aliases/index.js]: We can remove the `allowEditing` prop from the type too.

File: root/components/Aliases/index.js [hunk 2]
```diff
@@ -15,14 +15,30 @@ import EntityLink from '../../static/scripts/common/components/EntityLink';
...
 
+function canEdit($c: CatalystContextT, entityType: string) {
+  if ($c.user) {
+    switch (entityType) {
+      case 'area':
+        return $c.user.is_location_editor;
+      case 'instrument':
+        return $c.user.is_relationship_editor;
+      default:
+        return !$c.user.is_editing_disabled;
```
Comment (mwiencek) [inline root/components/Aliases/index.js]: I think it'd be more logically correct to `return true;` here and change the `if` to `if ($c.user && !$c.user.is_editing_disabled) {`.

**Comment by reosarevok** [comment]
Changes made, plus added a small extra commit fixing too long lines in the alias components

**Comment by reosarevok** [comment]
Updated - take a quick look to confirm before I merge maybe? :)

**Comment by yvanzo** [comment]
Confirmed!

**Pull request merged**